runtime/lua/vim/_defaults.lua

```diff
@@ -577,4 +577,6 @@ do
     vim.o.grepprg = 'rg --vimgrep -uu '
     vim.o.grepformat = '%f:%l:%c:%m'
   end
+
+  vim.o.splitright = true
```

Comment:
Change the default value in options.lua

https://github.com/neovim/neovim/blob/9afa1fd35510c5fe485f4a1dfdabf94e5f051a1c/src/nvim/options.lua#L7993